### PR TITLE
Remove allmost all deprecation from the files_version app

### PR DESCRIPTION
**Pull request by CarlSchwan**
The remaining deprecations are related to Utils::hooks and I will take a
look at how EventDispatcher works before working on them.

Aside from the deprecations, this patch also does a few minor
improvements around type hinting.

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

File: apps/files_versions/lib/AppInfo/Application.php [hunk 1]
```diff
@@ -125,7 +130,7 @@ private function loadBackend(array $backend, ContainerInterface $container, ILog
 			$backendObject = $container->get($class);
 			$versionManager->registerBackend($for, $backendObject);
 		} catch (\Exception $e) {
-			$logger->logException($e);
+			$logger->critical($e);
```
Comment (ChristophWurst) [inline apps/files_versions/lib/AppInfo/Application.php:133]:
```suggestion
			$logger->errpr($->getMessage(), ['exception' => $e]);
```

then it's equivalent to `logException($e)`

File: apps/files_versions/lib/Command/Expire.php [hunk 2]
```diff
@@ -65,11 +71,7 @@ public function handle() {
 			// In case of external storage and session credentials, the expiration
 			// fails because the command does not have those credentials
 
-			/** @var ILogger $logger */
-			$logger = \OC::$server->get(ILogger::class);
-
-			$logger->logException($e, [
-				'level' => ILogger::WARN,
+			$this->logger->warning($e, [
```
Comment (ChristophWurst) [inline apps/files_versions/lib/Command/Expire.php:74]:
```suggestion
			$this->logger->warning($e->getMessage, [
				'exception' => $e,
```

**Review by ChristophWurst — COMMENTED** [review]
careful with the old logger an exceptions. the PSR logger doesn't take exceptions as first argument

File: apps/files_versions/lib/Command/Expire.php [hunk 3]
```diff
@@ -65,11 +63,10 @@ public function handle() {
 			// In case of external storage and session credentials, the expiration
 			// fails because the command does not have those credentials
 
-			/** @var ILogger $logger */
-			$logger = \OC::$server->get(ILogger::class);
-
-			$logger->logException($e, [
-				'level' => ILogger::WARN,
+			/** @var LoggerInterface */
```
Comment (ChristophWurst) [inline apps/files_versions/lib/Command/Expire.php:66]:
```suggestion
			/** @var LoggerInterface $logger */
```

**Review by ChristophWurst — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**